clients/fides-js/src/components/Overlay.tsx

```diff
@@ -96,12 +97,23 @@ const Overlay: FunctionComponent<Props> = ({
     }
   }, [options, onOpen]);
 
+  const showBanner = useMemo(
+    () =>
+      !options.fidesDisableBanner &&
+      experience.experience_config?.component !== ComponentType.MODAL &&
```

Comment:
the key takeaways here are:
- `show_banner` will continue to be exposed as a computed property on the experience model (in the API response) for backward compatibility reasons only. it will now simply be based on the component type (i.e. whether it is `banner and modal` rather than just `modal`)
    - our updated FE code shouldn't be relying on `show_banner`, but should just be looking at the component type, and this is exactly what this code is doing 👍 
    - `show_banner` stays in the API response **solely for backward compatibility purposes**!
- this is an important attribute for customers and their privacy experiences. so we need to take care that we migrate its state accordingly, no matter the migration path we take
    - the most important piece of this is ensuring that experiences which previously had `banner_enabled = "always_disabled"` need to be migrated to an experience config that has `component = "modal"`, i.e. the banner continues to no longer show!

Reply:
My understanding is that we needed to migrate in accordance with [get_should_show_banner](https://github.com/ethyca/fides/blob/68c204a0b4ae716e6a3fe6e1e1fc5c2d84048561/src/fides/api/models/privacy_experience.py#L230), which originally populated that `show_banner` field.  It takes into account more than just the `banner_enabled` field. The worst part being whether there were privacy notices with various consent mechanisms in certain cases!  


Some more links @adamsachs 

- My comment on your [multilang migration doc](https://ethyca.atlassian.net/wiki/spaces/EN/pages/2918907921/Multilang+migrations+memo?focusedCommentId=2921103389)
- Some more clarifications [between Michael and I](https://ethyca.atlassian.net/wiki/spaces/PM/pages/2905997363?focusedCommentId=2906783815)

re: Current `show_banner` property, yes that's exactly what it's for, it was just added for backwards compatibility

Reply:
Yup, confirming the above. 

* `show_banner` should still be returned by BE purely for backwards compatibility as we can't guarantee all Fides clients will be upgraded along with our BE deploy.
* FE can remove dependence on `show_banner` an instead rely on the fact that BE only surfaces components that should show in the UI.